### PR TITLE
Heroku detection works on /etc/dyno

### PR DESCRIPTION
**Pull request by nateberkopec**

### CODE REVIEW

File: lib/raven/configuration.rb [hunk 1]
```diff
@@ -267,15 +267,23 @@ def project_root=(root_dir)
...
     end
 
     private
 
     def detect_release_from_heroku
-      ENV['HEROKU_SLUG_COMMIT']
+      sys_dyno_info = `cat /etc/heroku/dyno` rescue nil
+      return unless sys_dyno_info && sys_dyno_info != ""
+
+      begin
```
Comment (nateberkopec) [inline lib/raven/configuration.rb]: being overly cautious here because failure means Raven fails to start

**Comment by nateberkopec** [comment]
Bummer that Heroku users have to enable a labs extension to get release detection to work. #563

**Comment by gudmundur** [comment]
The plan is that this information becomes a default. The question, afaik, is how it'll be delivered.

**Comment by nateberkopec** [comment]
@gudmundur 👍  Where's the best place for me to keep to up-to-date on platform-level changes like that?

**Pull request merged**